fennel/datasets/datasets.py

```diff
@@ -723,6 +729,50 @@ def dsschema(self):
         return input_schema
 
 
+class Window(_Node):
+    def __init__(
+        self,
+        node: _Node,
+        keys: List[str],
+        type: str,
+        gap: Duration,
+        field: str,
+    ):
+        super().__init__()
+        if len(keys) == 0:
+            raise ValueError("Must specify at least one key")
```

Comment:
Error should be better - at least mention this is for window operator. 
Let's also add a validation to ensure that only `tumble`, `sliding`, `session` are allowed. Any other window types are invalid. Further, for `tumble`, `sliding` we should raise a different error that these windows aren't implemented yet.

Reply:
Will refractor this.

Reply:
Got it.